tests/foreman/api/test_contentview.py

```diff
@@ -112,6 +112,104 @@ def test_subscribe_system_to_cv(self):
         self.assertEqual(system_attrs['environment']['id'], lifecycle_env.id)
         self.assertEqual(system_attrs['organization_id'], org.id)
 
+    def test_cv_clone_within_same_env(self):
+        """
+        @Test: attempt to create, publish and promote new content view
+        based on existing view within the same environment as the
+        original content view
+
+        @Feature: Content Views
+
+        @Assert: Cloned content view can be published and promoted
+        to the same environment as the original content view
+
+        """
+        org = entities.Organization()
+        org.id = org.create()['id']
+        lifecycle_env = entities.LifecycleEnvironment(organization=org.id)
+        lifecycle_env.id = lifecycle_env.create()['id']
+        content_view = entities.ContentView(organization=org.id)
+        content_view.id = content_view.create()['id']
+
+        response = content_view.publish()
+        humanized_errors = response['humanized']['errors']
+        _check_bz_1186432(humanized_errors)
+        self.assertEqual(response['result'], 'success', humanized_errors)
+
+        cvv = entities.ContentViewVersion(
+            id=content_view.read_json()['versions'][0]['id']
+        )
+        response = cvv.promote(lifecycle_env.id)
+        humanized_errors = response['humanized']['errors']
+        _check_bz_1186432(humanized_errors)
+        self.assertEqual(response['result'], 'success', humanized_errors)
+
+        response = content_view.copy(gen_string('alpha', gen_integer(3, 30)))
```

Comment:
By random choosing a length this will ensure a more variable set of data being tested, I like the idea.